scripts/nlp_language_modeling/preprocess_data_for_megatron.py

```diff
@@ -152,12 +162,19 @@ def get_args():
 def main():
     args = get_args()
     startup_start = time.time()
-
-    print("Opening", args.input)
-    if args.input.endswith('.gz'):
-        fin = gzip.open(args.input, 'r')
+    if args.preproc_folder:
+        print('Searching folder for .json or .json.gz files...')
+        assert os.path.exists(args.input), 'Folder does not exist'
+        files_in_folder = os.listdir(args.input)
+        json_files = [
+            os.path.join(args.input, f) for f in files_in_folder if f.endswith('.json') or f.endswith('.json.gz')
+        ]
+        if len(json_files) == 0:
+            raise FileNotFoundError('No .json or .json.gz files found in folder.')
+        else:
+            print(f'Found {len(json_files)} .json or .json.gz files.')
     else:
-        fin = open(args.input, 'r', encoding='utf-8')
+        json_files = [args.input]
```

Comment:
might as well check the path exists here?

Reply:
Added a check here as well. Good catch.